cosmos/providers/dbt/render.py

```diff
@@ -111,12 +113,22 @@ def render_project(
             else:
                 run_args["outlets"] = outlets
 
-        # make the run task
-        run_task = Task(
-            id=f"{model_name}_run",
-            operator_class="cosmos.providers.dbt.core.operators.DbtRunOperator",
-            arguments=run_args,
-        )
+        if model.type == DbtModelType.DBT_MODEL:
+            # make the run task for model
+            run_task = Task(
+                id=f"{model_name}_run",
+                operator_class="cosmos.providers.dbt.core.operators.DbtRunOperator",
+                arguments=run_args,
+            )
+        elif model.type == DbtModelType.DBT_SNAPSHOT:
```

Comment:
@szecsip I think it's safe to assume that people might name their snapshot models the same as their base model (based on dbt's docs [here](https://docs.getdbt.com/docs/build/snapshots#example) they name the snapshot `orders` just like the base model `orders`). With the logic that you are proposing here, if there was an `orders` dbt model, then it would get overridden by the `orders` dbt snapshot. 

To add a visualization see what happens when I add `orders.sql` to my `/snapshots` directory:

```sql
{% snapshot orders_snapshot %}

{{
    config(
      target_schema='sandbox_chronek',
      unique_key='id',
      strategy='timestamp',
      updated_at='updated_at',
    )
}}

select * from {{ source('jaffle_shop', 'orders') }}

{% endsnapshot %}
```

![Screenshot 2023-03-27 at 10 58 33 AM](https://user-images.githubusercontent.com/31361051/228012743-191b84b0-4c38-47e8-ad41-f06aa6c415a3.png)

when `jaffle_shop` looked like this previously:

![Screenshot 2023-03-27 at 11 01 14 AM](https://user-images.githubusercontent.com/31361051/228013166-5ac459d4-275e-4073-9e5a-29b9be982e34.png)



**Can we change this so that both the dbt model and snapshot will parse if they are named the same thing?**

Reply:
@pohek321 Thank you for the quick testing. It sounds weird to named a modell and snapshot to same, but I could reproduce the issue. I try to find a solution which can handle this case too.

Reply:
@szecsip your solution is excellent! Was just telling @jlaneve that it's a really good pattern to follow. I did however get it to fail when changing the `test_behavior` parameter to `after_each` (see more details [here](https://astronomer.github.io/astronomer-cosmos/dbt/configuration.html#testing)). Mainly because cosmos got confused and tried to add a `test` step after the `snapshot`.

<img width="1058" alt="Screenshot 2023-04-07 at 3 37 26 PM" src="https://user-images.githubusercontent.com/31361051/230682019-e63a9423-fad0-49cf-ba25-bcab43948e1e.png">



I'll put together a commit so that your changes would work with `test_behavior` set to `after_all` or `after_each`

Reply:
@szecsip can I get a 👍/👎 on the latest commit [here](https://github.com/astronomer/astronomer-cosmos/pull/200/commits/98f3414db234f4ba2087ba1a2c3e192b02908048). This will (hopefully) resolve the previous comment that I made above 👆

Reply:
@pohek321 Thank you really much for the quick fix implementation. I would also do some really similar things.